genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -2914,9 +2973,839 @@ def support_driver(self, direction, i_g, i_b, i_o, shrink_sphere):
+            #     j = i - 1
+            #     px = -1.0 if (j & 1) == 0 else 1.0
+            #     py = -1.0 if (j & 2) == 0 else 1.0
+            #     pz = -1.0 if (j & 4) == 0 else 1.0
+
+            # n_dir = (dir + (gs.ti_vec3(px, py, pz) * gs.EPS)).normalized()
+
+            n_dir = dir
+            if i > 0:
+                j = i - 1
+                n_dir[0] += -gs.EPS if (j & 1) == 0 else gs.EPS
+                n_dir[1] += -gs.EPS if (j & 2) == 0 else gs.EPS
+                n_dir[2] += -gs.EPS if (j & 4) == 0 else gs.EPS
+
+            num_supports = self.func_count_support(i_ga, i_gb, i_b, n_dir)
+            if i > 0 and num_supports > 1:
+                # If this is a perturbed direction and we have more than one support point, we skip this iteration. If
+                # it was the original direction, we continue to find the support points to keep it as the baseline.
+                continue
+
+            # Use the current direction to find the support points.
+            for j in range(2):
+                d = n_dir if j == 0 else -n_dir
+                i_g = i_ga if j == 0 else i_gb
+
+                sp, si = self.support_driver(d, i_g, i_b, j, False)
+                if j == 0:
+                    obj1 = sp
+                    id1 = si
+                else:
+                    obj2 = sp
+                    id2 = si
+
+            mink = obj1 - obj2
+
+            if i == 0:
+                if num_supports > 1:
+                    # If there were multiple valid support points, we move on to the next iteration to perturb the
+                    # direction and find better support points.
+                    continue
+                else:
+                    break
+
+            # If it was a perturbed direction, check if the support points have been found before.
+            if i == 8:
+                # If this was the last iteration, we don't check if it has been found before.
+                break
+
+            # Check if the updated simplex would be a degenerate simplex.
+            if self.func_is_new_simplex_vertex_valid(i_b, id1, id2, mink):
+                break
+
+        return obj1, obj2, id1, id2, mink
+
+    @ti.func
+    def count_support_driver(self, d, i_g, i_b):
+        """
+        Count the number of possible support points in the given direction.
+        """
+        geom_type = self._solver.geoms_info[i_g].type
+        count = 1
+        if geom_type == gs.GEOM_TYPE.BOX:
+            count = self.support_field._func_count_supports_box(d, i_g, i_b)
+        elif geom_type == gs.GEOM_TYPE.MESH:
+            count = self.support_field._func_count_supports_world(d, i_g, i_b)
+        return count
+
+    @ti.func
+    def func_count_support(self, i_ga, i_gb, i_b, d):
+        """
+        Count the number of possible pairs of support points on the two objects in the given direction [d].
+        """
+        count = 1
+        for i in range(2):
+            dir = d if i == 0 else -d
```

Comment:
Just inline those arguments: `count *= self.count_support_driver(d if i == 0 else -d, i_ga if i == 0 else i_gb, i_b)`

And rename the input argument `d` in `dir`, it would be more clear.